wasmtime/_func.py

```diff
@@ -56,6 +63,36 @@ def type(self, store: Storelike) -> FuncType:
         ptr = ffi.wasmtime_func_type(store._context, byref(self._func))
         return FuncType._from_ptr(ptr, None)
 
+    def _create_raw_vals(self, *params: IntoVal) -> ctypes.Array[wasmtime_val_raw_t]:
+        raw = self._vals_raw_type()
+        for i, param_str in enumerate(self._params_str):
+            setattr(raw[i], param_str, params[i])
```

Comment:
I don't think that this conversion is quite correct. The `IntoVal` type could also be `Val`, for example, to handle things like v128/funcref/externref/etc. In such a case the raw attribute of `raw[i]` can't be set to that value, and I'm not sure what happens otherwise during such a process.

Additionally it's not safe to simply set the `externref` and `funcref` fields. Those need to go through the methods in the `Val` type.

Reply:
For an example you can see the logic in `Value._convert`